supplier/views.py

```diff
@@ -1,15 +1,49 @@
 from django.shortcuts import render
 from supplier.models import Supplier
 from django.contrib.auth.decorators import user_passes_test
+from supplier_product.models import SupplierProduct
+from product.models import Product
+from django.contrib import messages
 
 
-def supplier_check(user):
+def get_supplier(user):
     try:
         return Supplier.objects.get(supplier_account=user)
     except Exception:
         return False
 
 
-@user_passes_test(supplier_check, login_url='Not allowed')
+def receive_product(request, supplier):
+    prod_name = request.GET['searched_product']
+    if prod_name:
+        product = Product.filter_name(prod_name).first()
+        supplier_product = SupplierProduct.objects.filter(qr_code=product,
+                                                          user_name=supplier)
+        if supplier_product.exists():
+            return supplier_product
+    return None
+
+
+def action_searched_product(request, supplier):
```

Comment:
What does this function suppose to do? Usually, function names should be some sort of action (verb).

Reply:
the thought was to make a dictionary of actions for the supplier page, but i assume you were talking about the 'searched' product, so yea i can agree on that, will change it to search product